src/Time/TimeSteppers/RungeKutta3.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class RungeKutta3.
+
+#pragma once
+
+#include <deque>
+#include <string>
+
+#include "ErrorHandling/Error.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeSteppers/TimeStepper.hpp"
+
+namespace TimeSteppers {
+
+/// A "strong stability-preserving" 3rd-order Runge-Kutta time-stepper.
+/// Major reference:  J. Hesthaven & T. Warburton, Nodal Discontinuous
+/// Galerkin Methods. section 5.7
+class RungeKutta3 : public TimeStepper::Inherit {
+ public:
+  //using OptionsList = tmpl::list<>;
+  //static std::string class_id() { return "RungeKutta3"; }
+  //static constexpr OptionString_t help = {
+  //    "A third-order strong stability-preserving Runge-Kutta time-stepper."};
+
+  RungeKutta3() = default;
+  RungeKutta3(const RungeKutta3&) = default;
+  RungeKutta3& operator=(const RungeKutta3&) = default;
+  RungeKutta3(RungeKutta3&&) = default;
+  RungeKutta3& operator=(RungeKutta3&&) = default;
+  ~RungeKutta3() override = default;
+
+  template <typename Vars, typename DerivVars>
+  TimeDelta update_u(
+      Vars& u,
+      const std::deque<std::tuple<Time, Vars, DerivVars>>& history,
+      const TimeDelta& time_step) const;
+
+  size_t number_of_substeps() const noexcept override;
+
+  size_t number_of_past_steps() const noexcept override;
+
+  bool is_self_starting() const noexcept override;
+
+  double stable_step() const noexcept override;
+};
+
+template <typename Vars, typename DerivVars>
+TimeDelta RungeKutta3::update_u(
+    Vars& u,
+    const std::deque<std::tuple<Time, Vars, DerivVars>>& history,
+    const TimeDelta& step_size) const {
+  const size_t substep = history.size() - 1;
+  const auto& vars = std::get<1>(history.back());
+  const auto& dt_vars = std::get<2>(history.back());
+  const auto& U0 = std::get<1>(history[0]);
+
+  switch (substep) {
+    case 0: {
+      // from (5.32) of Hesthaven
```

Comment:
I think these would be hard to move out-of-line and have them be as useful, since they are mostly explaining what the states of the variables are at different points in the function.